plugins/module_utils/haproxy_frontend_server.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Chris Morton, cosmo@cosmo.2y.net
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+import re
+from ansible_collections.pfsensible.core.plugins.module_utils.module_base import PFSenseModuleBase
+
+HAPROXY_FRONTEND_SERVER_ARGUMENT_SPEC = dict(
+    state=dict(default='present', choices=['present', 'absent']),
+    frontend=dict(required=True, type='str'),
+    extaddr=dict(required=True, type='str'),
+    extaddr_port=dict(required=True, type='int'),
+    extaddr_ssl=dict(required=True, type='str'),
```

Comment:
You really don't want anything other name the id field to be required, otherwise you break removal with just the id.  Use required_if with state present.